pkg/model/manifest.go

```diff
@@ -722,6 +722,8 @@ func (m *Manifest) setDefaults() error {
 			b.Context = b.Name
 			b.Name = ""
 		}
+		b.setDockerfileDefaults()
+
 		if !(b.Image != "" && len(b.VolumesToInclude) > 0 && b.Dockerfile == "") {
```

Comment:
As we are setting the default for `b.Dockerfile` in line 725, should we remove the `b.Dockerfile == ""` from this conditional as it will always contain a value now (and provokes that the code within the if is always executed)?

Reply:
But if the context is empty, nothing happens in the DockerfileDefaults, and it has to continue making that condition

Reply:
oh, that's true. This is confusing and very error prone. Agree with Jacob on find a way to refactor and simplify all this logic